src/planning/planner.ts

```diff
@@ -114,10 +114,7 @@ function _validateActiveBindingCount(
             }
 
         case BindingCount.OnlyOneBindingAvailable:
-            if (!target.isArray()) {
-                return bindings;
-            }
-
+            return bindings;
```

Comment:
we still need to throw something here

Reply:
Yeah, we need to investigate that case

Reply:
@PodaruDragos I investigated this case in an issue, I think this code is right, we should allways return the bindings, see [this comment](https://github.com/inversify/InversifyJS/issues/1271#issuecomment-822444015) for more information